src/frontend/src/App.js

```diff
@@ -45,6 +47,20 @@ function App() {
       disable_ocean_mask: formRefs.disableOceanMask.current.checked,
     };
 
+    try {
```

Comment:
1. this piece of code occurs in 3 different places. let's try to move it to a separate function and call it where we want to check the taxa name.

2. do we really need to perform this check 3 times?

Reply:
> 2.

Yes for generating, saving, clearing. If taxa is empty or invalid, generating crashes the server. Same way saving also crashes because it cannot parse the taxa name correctly. Clear also makes a request to save empty annotation, same reason for crash